src/Database/Type/DateTimeFractionalType.php

```diff
@@ -0,0 +1,40 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database\Type;
+
+/**
+ * Extends DateTimeType with support for fractional seconds up to microseconds.
+ */
+class DateTimeFractionalType extends DateTimeType
+{
+    /**
+     * @inheritDoc
+     */
+    protected $_format = 'Y-m-d H:i:s.u';
+
+    /**
+     * @inheritDoc
+     */
+    protected $_marshalFormats = [
```

Comment:
Going through the PHP manual page for [DatetimeInterface](https://www.php.net/manual/en/class.datetimeinterface.php) I came across the constant `DateTimeInterface::RFC3339_EXTENDED` who's format string has placeholder `v` for milliseconds.

So should we included formats with `v` too in this list?

Reply:
So `v` is milliseconds only.  The reason I switched to createFromFormat() instead of comparing strings is to allow 1-6 digits as you can't really control how many digits the database will return or how many digits a user needs to include.

I hope to remove the strict formatting restrictions in the future.